core/system/src/page_allocator.c

```diff
@@ -0,0 +1,272 @@
+/*
+ * Copyright (c) 2016, ARM Limited, All Rights Reserved
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* This file can be compiled externally to provide the page allocator algorithm
+   for devices NOT supported by uVisor. For this purpose this file is copied as
+   is into uvisor-mbed-lib and compiled by the target build system. */
+#if defined(UVISOR_PRESENT) && (UVISOR_PRESENT == 1)
+#include <uvisor.h>
+#include "page_allocator.h"
```

Comment:
How does it work for `page_allocator.h`? It must be copied as well? Other header files (`*_exports.h` ones) are usually available for free (or copied altogether), but this would need to be copied separately.